src/main/java/net/dries007/tfc/common/blockentities/rotation/TripHammerBlockEntity.java

```diff
@@ -43,11 +45,30 @@ public static void serverTick(Level level, BlockPos pos, BlockState state, TripH
         if (rotation != null)
         {
             final float angle = hammer.getRealRotationDegrees(rotation, 1f);
-            if (angle > 180f && angle < 183f)
+            // Guards against:
+            // 1. the angle wrapping around from 360 to 0
+            // 2. the rotation speed being too fast and/or offset enough to sneak past the expected angle
+            if (angle > 180 && hammer.lastAngle < 180 && hammer.lastAngle > 90)
             {
                 if (rotation.positiveDirection() != state.getValue(TripHammerBlock.FACING).getClockWise())
                 {
-                    level.destroyBlock(pos, true);
+                    ItemStack droppedItem = hammer.inventory.extractItem(0, 1, false);
+                    if (droppedItem.isDamageableItem())
+                    {
+                        droppedItem.hurtAndBreak(droppedItem.getMaxDamage() / 4 + 1, (ServerLevel) level, null, i -> {});
+                        if (!droppedItem.isEmpty())
+                        {
+                            Helpers.spawnItem(level, pos, droppedItem);
+                        }
+                        else
+                        {
+                            level.playSound(null, pos, SoundEvents.ITEM_BREAK, SoundSource.BLOCKS);
+                        }
+                    }
+                    // If the item is not damageable it gets shredded. Sorry.
```

Comment:
Is there anything in base TFC that isn't damageable that could be used in a trip hammer? Because if this case doesn't exist in vanilla TFC, then it seems like we should drop the item because the most likely use case I can think of is some modpack that wants to give you an unbreakable upgrade to the trip hammer

Reply:
Everything that goes in the trip hammer is damageable. This is a good point however